nasa-gibs/worldview
# WV-2497: Updating Granule Count for Layers Under Same Satellite

## Description

This updates our granule count logic to update the granule counts for each granule layer that shares a parent satellite when one granule layer count is updated. 

I am not entirely sure on how to set up our granule layers without merging the `granule-configs` [https://github.com/nasa-gibs/worldview/commit/a784a54949807f7b8583c836d27da44554476b59](url) branch into my own. Since I merged this branch it will be easy enough to test since you will not have to perform any extra steps to access the granule layers. But ultimately when this PR is approved, I will have to create a separate branch and copy my changes there without the granule layers changes to the config/build files.

## How To Test

1. `git checkout wv-2497-granule-count`
2. `npm ci`
3. `npm run build`
4. `npm run watch`
5. Use this URL [http://localhost:3000/?v=-457.64927151032043,-263.3386798522034,152.10072848967957,219.84882014779657&z=4&i=4&ics=true&ici=5&icd=101&l=Coastlines_15m,BlueMarble_NextGeneration(hidden),VIIRS_NOAA20_CorrectedReflectance_TrueColor(hidden),Land_Water_Map,OrbitTracks_NOAA-20_Ascending(hidden),VIIRS_NOAA20_CorrectedReflectance_BandsM3-I3-M11_Granule(count=11),VIIRS_NOAA20_CorrectedReflectance_BandsM11-I2-I1_Granule(hidden,count=12),VIIRS_NOAA20_CorrectedReflectance_TrueColor_Granule(count=11)&lg=false&t=2022-11-22-T16%3A10%3A00Z](url)
6. Verify that the granule layers are loaded and visible.
7. Click on the options button in the sidebar layer row for one of the granule layers.
8. Verify that the new tooltip next to the `Granule Count` header reads `Updating granule count for all granules layers associated with the NOAA-20/VIIRS satellite.`
9. Change the granule count with the slider and note the count that you changed to.
10. Verify that the map has re-rendered.
11. Verify that each of the other 2 granule layers now have the same granule count as what you changed the original one to.
